### PR TITLE
[Fad 83] [FAD-125] [FAD-128] Launch pool and approval endpoint

### PR DESCRIPTION
**Pull request by v1n33th**
### Description
- Added smart contract creation on launch pool button click
- Added approval endpoint for approving in-review pools (api/pool/[poolId]) patch request
- Added rest API support for apollo client
- added pending pools section in artist dashboard
-  added filtering to pool API

- [ ] `CHANGELOG` has been updated (_if appropriate_)
- [ ] Environment variables updated (_if appropriate_)

### CODE REVIEW

File: packages/client3/components/dashboard/pool-card/pending-pool-card.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        <Caption className="pt-[3px] text-dark-50">{poolName}</Caption>
+      </div>
+      <>
+        <div className="flex-1 pt-[28px] text-center capitalize">
+          <Chip
+            type={
+              statusType == Pool_Status_Type.InReview ? "pending" : "completed"
+            }
+          >
+            {statusType == Pool_Status_Type.InReview ? "In Review" : "Approved"}
```
Comment (sachasmart-weavik) [inline packages/client3/components/dashboard/pool-card/pending-pool-card.tsx:55]: I think I had to mess around with these in [FAD-138] to make sure they were accurate.

File: packages/client3/lib/graphql/rest-api-schema.graphql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  id: String!
+  poolName: String!
+  projectDetail: String!
+  goalAmount: String!
+  closingDate: String!
+  projectCoverImage: String
+  walletAddress: String!
+  terms: Terms
+  status: POOL_STATUS_TYPE!
+  transactionHash: String
```
Comment (sachasmart-weavik) [inline packages/client3/lib/graphql/rest-api-schema.graphql:11]: This is addressed in 138 but `transactionHash` to `poolAddress` 

Either we can leave it and change it later with 138 or do it now.

File: packages/client3/pages/api/pool/index.page.ts [hunk 3]
```diff
@@ -2,61 +2,125 @@ import fs from "fs";
...
+        _hasTransactionHash.toString().toLowerCase() == "true";
+      if (hasTransactionHash && !pool.transactionHash) {
+        return false;
+      }
+      if (!hasTransactionHash && !!pool.transactionHash) {
+        return false;
+      }
+    }
+    return true;
+  });
```
Comment (sachasmart-weavik) [inline packages/client3/pages/api/pool/index.page.ts:54]: Again now or later with `transactionHash`

File: packages/client3/pages/artist/dashboard/pending-pools-artist.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    const receipt = await createPool(goldfinchFactory, pool.goalAmount);
+    await updatePoolTransactionHash(pool.id, receipt);
+    refetch();
+  };
+  const handleClick = (poolAddress: string) => {
+    /**
+     * TODO: Pool details screen requires some refactoring to handle pending pool
+     */
+    //router.push(`/artist/pool/${poolAddress}`);
+  };
```
Comment (sachasmart-weavik) [inline packages/client3/pages/artist/dashboard/pending-pools-artist.tsx:67]: We should probably make a ticket for this

Comment (sachasmart-weavik) [reply packages/client3/pages/artist/dashboard/pending-pools-artist.tsx:67]: Done 
https://weavik.atlassian.net/browse/FAD-142

**Review by sachasmart-weavik — APPROVED** [review]
Nice work my man. 

Comments and thoughts only

File: packages/client3/pages/artist/dashboard/pending-pools-artist.tsx (hunk 4 above)
Comment (v1n33th) [reply packages/client3/pages/artist/dashboard/pending-pools-artist.tsx:67]: Thanks man

File: packages/client3/lib/graphql/rest-api-schema.graphql (hunk 2 above)
Comment (v1n33th) [reply packages/client3/lib/graphql/rest-api-schema.graphql:11]: Have ya changed it in 138?

Comment (sachasmart-weavik) [reply packages/client3/lib/graphql/rest-api-schema.graphql:11]: Yes

Comment (v1n33th) [reply packages/client3/lib/graphql/rest-api-schema.graphql:11]: Awesome thank you...Then we will add related changes there

**Pull request merged**